lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -1494,7 +1481,7 @@ txConstraints era protocolParams witnessTag = TxConstraints
     -- skeleton.
     marginalCostOf :: TxSkeleton -> Coin
     marginalCostOf =
-        Coin.distance txBaseCost . estimateTxCost era protocolParams
+        Coin.distance txBaseCost . estimateTxCost era feePerByte
```

Comment:
I'm just wondering whether `Coin.distance` is really the right function to use here.

The `Coin.distance` function computes the **_absolute difference_**, **_regardless_** of which amount is higher. In other words `∀ a b. Coin.distance a b == Coin.distance b a`.

I suspect what we really want to find is the **_excess_** of some amount `a` over amount `b`, where we believe amount `a` is higher.

In that case, perhaps we should use `Coin.difference` here, which performs truncated subtraction.

Reply:
Leaving as is, but created https://input-output.atlassian.net/browse/ADP-3016 for addressing